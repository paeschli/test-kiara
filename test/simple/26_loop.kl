extern [C] include_file_ctx(ctx:ptr(Context), fileName:ptr(char)):void;
def include_file(fileName:ptr(char)) include_file_ctx(getContext(), fileName);

include_file("stdlib.kl");

def intrinsic [llvm] binary + 20 (a:int, b:int):int
"define %int @${mangledName}(%int %a, %int %b) nounwind uwtable readnone {"
"  %add = add nsw i32 %b, %a"
"  ret %int %add"
"}";

def intrinsic [llvm] binary > 10 (a:int, b:int):boolean
"define zeroext i1 @${mangledName}(i32 %a, i32 %b) nounwind uwtable readnone {"
"  %cmp = icmp sgt i32 %a, %b"
"  ret i1 %cmp"
"}";

def test() {
    var i:int in {
      i = 0;

      stop1: {
            loop stop2: {
                 print("iter\n");
                 break stop1;
                 i = i + 1;
                 print(i);
                 print("\n");
            }
      }

      stop3: {
            loop stop4: {
                 if (i > 9) then
                    break stop3;
                 print("iter ");
                 print(i);
                 print("\n");
                 i = i + 1;
            }
      }
    }
}

test();
