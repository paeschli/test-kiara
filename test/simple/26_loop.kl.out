iter
iter 0
iter 1
iter 2
iter 3
iter 4
iter 5
iter 6
iter 7
iter 8
iter 9
